plumtree-servlet/src/main/java/org/apache/tuweni/plumtree/servlet/GossipServlet.java

```diff
@@ -0,0 +1,254 @@
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GossipServlet extends HttpServlet {
+
+  private static final String PLUMTREE_SERVER_HEADER = "Plumtree-Server";
+
+  private static final Logger logger = LoggerFactory.getLogger(GossipServlet.class);
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  private static final class Message {
+
+    public MessageSender.Verb verb;
+    public String attributes;
+    public String hash;
+    public String payload;
+  }
+
+  private final int graftDelay;
+  private final int lazyQueueInterval;
+  private final MessageHashing messageHashing;
+  private final String networkInterface;
+  private final int port;
+  private final MessageListener payloadListener;
+  private final MessageValidator payloadValidator;
+  private final PeerPruning peerPruningFunction;
+  private final PeerRepository peerRepository;
+
+  private final AtomicBoolean started = new AtomicBoolean(false);
+  private State state;
+
+  public GossipServlet(
+      int graftDelay,
+      int lazyQueueInterval,
+      MessageHashing messageHashing,
+      String networkInterface,
+      int port,
+      MessageListener payloadListener,
+      MessageValidator payloadValidator,
+      PeerPruning peerPruningFunction,
+      PeerRepository peerRepository) {
+    this.graftDelay = graftDelay;
+    this.lazyQueueInterval = lazyQueueInterval;
+    this.messageHashing = messageHashing;
+    this.networkInterface = networkInterface;
+    this.port = port;
+    this.payloadListener = payloadListener;
+    this.payloadValidator = payloadValidator == null ? (bytes, peer) -> true : payloadValidator;
+    this.peerPruningFunction = peerPruningFunction == null ? (peer) -> true : peerPruningFunction;
+    this.peerRepository = peerRepository;
+  }
+
+  private CloseableHttpClient httpclient;
+
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    super.init(config);
+    if (started.compareAndSet(false, true)) {
+      httpclient = HttpClients.createDefault();
+      state = new State(peerRepository, messageHashing, (verb, attributes, peer, hash, payload) -> {
+        Message message = new Message();
+        message.verb = verb;
+        message.attributes = attributes;
+        message.hash = hash.toHexString();
+        message.payload = payload == null ? null : payload.toHexString();
+        HttpPost postMessage = new HttpPost("http://" + ((ServletPeer) peer).getAddress());
+        postMessage.setHeader(PLUMTREE_SERVER_HEADER, this.networkInterface + ":" + this.port);
```

Comment:
I guess this is a means to self seed the peers? How do they discover each other?
Would they need to be colocated in the same process?

Reply:
This is indeed how you tell the other peer how to communicate with you. Discovery is not part of plumtree.
You can see in the test how we connect the peers. If you want discovery, there are a few ways to go about it :) there are a few libs in Tuweni to deal with it.

Reply:
One of the options is to use plumtree to gossip new peers to all peers so they all connect, if you're interested.

Reply:
Interesting. I'm thinking that I may assume that there are certain seed nodes that are well known to each deployment that can bootstrap the initial set of peers and use gossip for that interaction to spread the new one across the others.